app/src/main/java/org/gnucash/android/export/csv/CsvTransactionsExporter.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Creates a GnuCash CSV transactions representation of the accounts and transactions
+ *
+ * @author Semyannikov Gleb <nightdevgame@gmail.com>
+ */
+public class CsvTransactionsExporter extends Exporter{
+
+    private char mCsvSeparator;
+
+    /**
+     * Construct a new exporter with export parameters
+     * @param params Parameters for the export
+     */
+    public CsvTransactionsExporter(ExportParams params) {
+        super(params, null);
+        mCsvSeparator = params.getCsvSeparator();
+        LOG_TAG = "GncXmlExporter";
+    }
+
+    /**
+     * Overloaded constructor.
+     * Creates an exporter with an already open database instance.
+     * @param params Parameters for the export
+     * @param db SQLite database
+     */
+    public CsvTransactionsExporter(ExportParams params, SQLiteDatabase db) {
+        super(params, db);
+        mCsvSeparator = params.getCsvSeparator();
+        LOG_TAG = "GncXmlExporter";
+    }
+
+    @Override
+    public List<String> generateExport() throws ExporterException {
+        OutputStreamWriter writerStream = null;
+        CsvWriter writer = null;
+        String outputFile = getExportCacheFilePath();
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(outputFile);
+            BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileOutputStream);
+            writerStream = new OutputStreamWriter(bufferedOutputStream);
+            writer = new CsvWriter(writerStream);
+            generateExport(writer);
+        } catch (IOException ex){
+            Crashlytics.log("Error exporting CSV");
+            Crashlytics.logException(ex);
+        } finally {
+            if (writerStream != null) {
+                try {
+                    writerStream.close();
+                } catch (IOException e) {
+                    throw new ExporterException(mExportParams, e);
+                }
+            }
+        }
+
+        List<String> exportedFiles = new ArrayList<>();
+        exportedFiles.add(outputFile);
+
+        return exportedFiles;
+    }
+
+    public void generateExport(final CsvWriter writer) throws ExporterException {
+        try {
+            String separator = mCsvSeparator + "";
+            List<String> names = new ArrayList<String>();
+            names.add("Date");
+            names.add("Account name");
+            names.add("Number");
+            names.add("Description");
+            names.add("Notes");
+            names.add("Memo");
+            names.add("Category");
+            names.add("Type");
+            names.add("Action");
+            names.add("Reconcile");
+            names.add("To With Sym");
+            names.add("From With Sym");
+            names.add("To Num.");
+            names.add("From Num.");
+            names.add("To Rate/Price");
+            names.add("From Rate/Price");
+
+            List<Transaction> transactions = mTransactionsDbAdapter.getAllTransactions();
```

Comment:
The number of transactions in the database can be quite big for very large books over time. It is not efficient to load everything into memory. It is better to use a cursor to iterate over the database and retrieve the entries you need on-demand. 
I know just getting all records is easier at first, but it will become technical debt in the future

Reply:
I'll try to fix it in 5-10 days. Now i have trouble because i have no big db. Can you help me and give some example of big enough book in xml? I need it to test export speed and compatibility with gnucash desktop

Reply:
Hmm, I do not have any big db. I've been thinking to write some modules in the test framework which generates tons of transactions. But I never got around to it.

No matter, I think you can just switch it to cursor iteration without having to actually have a big book to test it on. Users who have them will appreciate it.